### PR TITLE
Add tests for non existent page routes, check 404 exists

### PR DESCRIPTION
**Pull request by jakierice**

### CODE REVIEW

File: integration-tests/production-runtime/cypress/integration/production.js [hunk 1]
```diff
@@ -14,13 +14,23 @@ describe(`Production build tests`, () => {
...
     cy.waitForRouteChange()
       .getTestElement(`page2`)
       .should(`exist`)
       .location(`pathname`)
       .should(`equal`, `/`)
   })
+
+  it(`should show 404 page when visiting non-existent page route`, () => {
+    cy.getTestElement(`404`)
+      .first()
```
Comment (DSchau) [inline integration-tests/production-runtime/cypress/integration/production.js]: Shouldn't technically need `first` here but it's not hurting anything!

File: integration-tests/production-runtime/cypress/integration/production.js [hunk 2]
```diff
@@ -14,13 +14,23 @@ describe(`Production build tests`, () => {
...
+      .click()
+
+    cy.waitForRouteChange()
+      .getTestElement(`404`)
```
Comment (DSchau) [inline integration-tests/production-runtime/cypress/integration/production.js]: You could also [check the pathname here](https://github.com/gatsbyjs/gatsby/blob/master/integration-tests/path-prefix/cypress/integration/path-prefix.js#L25), too! But this works just as well!

Comment (jakierice) [reply integration-tests/production-runtime/cypress/integration/production.js]: @DSchau thanks for the heads up! We took the `.first()` call out and added assertion for pathname validity.

**Comment by KyleAMathews** [comment]
Thanks @jakierice!

**Pull request merged**